rust/agama-network/src/model.rs

```diff
@@ -220,28 +222,32 @@ impl NetworkState {
         controller: &Connection,
         ports: Vec<String>,
     ) -> Result<(), NetworkStateError> {
-        if let ConnectionConfig::Bond(_) = &controller.config {
-            let mut controlled = vec![];
-            for port in ports {
-                let connection = self
-                    .get_connection_by_interface(&port)
-                    .or_else(|| self.get_connection(&port))
-                    .ok_or(NetworkStateError::UnknownConnection(port))?;
-                controlled.push(connection.uuid);
-            }
+        match &controller.config {
+            ConnectionConfig::Bond(_) | ConnectionConfig::Bridge(_) => {
+                let mut controlled = vec![];
+                for port in ports {
+                    let connection = self
+                        .get_connection_by_interface(&port)
+                        .or_else(|| self.get_connection(&port))
+                        .ok_or(NetworkStateError::UnknownConnection(port))?;
+                    controlled.push(connection.uuid);
+                }
 
-            for conn in self.connections.iter_mut() {
-                if controlled.contains(&conn.uuid) {
-                    conn.controller = Some(controller.uuid);
-                } else if conn.controller == Some(controller.uuid) {
-                    conn.controller = None;
+                for conn in self.connections.iter_mut() {
+                    if controlled.contains(&conn.uuid) {
+                        conn.controller = Some(controller.uuid);
+                        if conn.interface.is_none() {
+                            conn.interface = Some(conn.clone().id);
```

Comment:
You do not need to clone the whole object.

```suggestion
                            conn.interface = Some(conn.id.clone());
```